Add recipe for olc

ookisan opened the pull request:
### Brief summary of what the package does

Encoding, decoding, shortening, and recovery of open location codes (a.k.a. "plus codes") for emacs lisp. Passes the "official" test cases, conforms to the requirements on libraries in the specification.

### Direct link to the package repository

https://gitlab.liu.se/davby02/olc.git

### Your association with the package

Author and maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Looks good, and thank you for your close adherence to the checklist.

A couple notes, but these aren't blocking:
- latitute -> latitude
- There's no need for `progn` inside individual cases in a `cond`
- `(and (and x y) z)` is the same as `(and x y z)` if you were inclined to modify the form in `olc-is-valid`
- 'compound' in `olc-is-short` could be capitalized (checkdoc occasionally misses)
- re: "Unfortunately check-declare can't handle these." you have a few options to silence `check-declare` -- e.g. `(declare-function request "ext: request" t t)`

And, welcome to MELPA!

ookisan commented:
Thanks for the feedback.

* latitute -> latitude: fixed.
* `progn` inside `cond`: remnnant of an old `if`; fixed.
* Nested `and`: unchanged; I find the complex condition easier to understand with the nesting intact.
* 'compound' -> 'COMPOUND' in docstring: fixed.
* re: `check-declare`: thanks; I learned something new from this.

riscy commented:
No worries. :)  I noticed a typo in my check-declare up there (added space).  Should be: `(declare-function request "ext:request" t t)` in the example.
